relay-common/src/utils.rs

```diff
@@ -1,11 +1,49 @@
 use std::fmt;
+use std::mem;
 use std::str;
+use std::sync::{Arc, Mutex};
 
 use once_cell::sync::OnceCell;
 use regex::Regex;
 
 use crate::macros::impl_str_serde;
 
+/// Wraps either the provided pattern string, which will be consumed and replaced with compiled
+/// regex, when the `compiled` function is called.
+#[derive(Debug, Clone)]
+struct GlobRegex {
+    // Note: Mutex and Arc here used to make sure this struct can be Send and Sync. Also that we
+    // could use interior mutabilty in order to be able to consume the value when needed.
+    value: Arc<Mutex<String>>,
```

Comment:
This doesn't have to be behind a mutex. In order to construct a `Regex`, you can simply borrow the existing string.

Reply:
Yeah, it's me wanted this to be with interior mutability and actually consume the string when the regex is compiled. But I guess it's ok to keep it around as well.